lib/Source.php

```diff
@@ -66,6 +66,41 @@ public function save($opts = null)
 
     /**
      * @param array|null $params
+     * @param array|string|null $opts
+     *
+     * @return Source The deleted source.
+     */
+    public function delete($params = null, $options = null)
+    {
+        self::_validateParams($params);
+
+        $id = $this['id'];
+        if (!$id) {
+            $class = get_class($this);
+            $msg = "Could not determine which URL to request: $class instance "
+             . "has invalid ID: $id";
+            throw new Error\InvalidRequest($msg, null);
+        }
```

Comment:
Yeah, I guess -- I mostly just copied the existing logic when deleting API resources: [`ApiResource::_delete`](https://github.com/stripe/stripe-php/blob/11c6041f7aea18905c121e46adf09dc4427cd14a/lib/ApiResource.php#L194) calls [`instanceUrl`](https://github.com/stripe/stripe-php/blob/11c6041f7aea18905c121e46adf09dc4427cd14a/lib/ApiResource.php#L91), which in turn calls [`resourceUrl`](https://github.com/stripe/stripe-php/blob/11c6041f7aea18905c121e46adf09dc4427cd14a/lib/ApiResource.php#L74) which has [this block](https://github.com/stripe/stripe-php/blob/11c6041f7aea18905c121e46adf09dc4427cd14a/lib/ApiResource.php#L76-L81).

For sources, we can't use the existing method because sources behave differently than existing customer payment sources (ugh naming), i.e. cards & bank accounts:
- their normal instance URL (for retrieve & update requests) is `/v1/sources/src_foo`
- sources cannot be deleted, but when attached to a customer, they can be detached by sending a `DELETE` request to `/v1/customers/cus_bar/sources/src_foo`.

If you want, I can refactor to avoid duplicating the code, or just get rid of the check entirely as it doesn't seem terriblye useful.